### ISSUE TITLE
[BUG] Ansible execution fails

### ISSUE THREAD

**Issue by peter-fe** [saltstack/salt#57938]
**Description**

Calling a simple Ansible system.ping on a minion fails.
```
salt-call --local ansible.system.ping

Passed invalid arguments: a bytes-like object is required, not 'str'.

Usage:
Available
```
 
**Setup**
```
CentOS Linux release 8.2.2004 (Core)
```
```
pip3 --version
pip 9.0.3 from /usr/lib/python3.6/site-packages (python 3.6)
```
```
python3 --version
Python 3.6.8
```
```
ansible --version
ansible 2.9.10
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.8 (default, Apr 16 2020, 01:36:27) [GCC 8.3.1 20191121 (Red Hat 8.3.1-5)]

```

**Steps to Reproduce the behavior**

Fails:
```
salt-call --local ansible.system.ping
```
Works:
```
salt-call --local ansible.list
local:
    - cloud.alicloud.ali_instance
    - cloud.alicloud.ali_instance_info
    - cloud.amazon._ec2_ami_find
    - cloud.amazon._ec2_ami_search
    - cloud.amazon._ec2_facts
    - cloud.amazon._ec2_remote_facts
    - cloud.amazon._ec2_vpc
    - cloud.amazon._lambda_facts
    - cloud.amazon._s3
    - cloud.amazon.aws_acm_info
    - cloud.amazon.aws_api_gateway
...
```
```
salt-call --local ansible.help system.ping
local:
    ----------
    Available sections on module "system.ping":
    Description:
        - A trivial test module, this module always returns C(pong) on successful contact. It does not make sense in playbooks, but it is useful from C(/usr/bin/ansible) to verify the ability to login and that a usable Python is configured.
        - This is NOT ICMP ping, this is just a trivial test module that requires Python on the remote-node.
        - For Windows targets, use the M(win_ping) module instead.
        - For Network targets, use the M(net_ping) module instead.
```
```
ansible localhost -m ping
[WARNING]: No inventory was parsed, only implicit localhost is available
localhost | SUCCESS => {
    "changed": false,
    "ping": "pong"
}
```

**Expected behavior**

A call to Ansible through Salt succeeds and delivers feedback

**Versions Report**
```
Salt Version:
           Salt: 3000.3

Dependency Versions:
           cffi: 1.11.5
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
         Jinja2: 2.10.1
        libgit2: Not Installed
       M2Crypto: 0.33.0
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.6.1
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: Not Installed
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.6.8 (default, Apr 16 2020, 01:36:27)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 17.0.0
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.3
            ZMQ: 4.3.1

System Versions:
           dist: centos 8.2.2004 Core
         locale: UTF-8
        machine: x86_64
        release: 4.18.0-193.6.3.el8_2.x86_64
         system: Linux
        version: CentOS Linux 8.2.2004 Core
```
</details>

**Comment by slinderud** [comment]
Any update on this?

Still broken.

```
# salt salt-master* state.apply test
salt-master:
----------
          ID: run_test_echo
    Function: ansible.call
      Result: False
     Comment:
     Started: 17:59:12.634493
    Duration: 1632.838 ms
     Changes:
              ----------
              system.ping:
                  Module "system.ping" failed. Error message: (KeyError) 'ansible.system.ping'

Summary for salt-master
------------
Succeeded: 0 (changed=1)
Failed:    1
------------
Total states run:     1
Total run time:   1.633 s
ERROR: Minions returned with non-zero exit code

# cat /srv/salt/test.sls
run_test_echo:
  ansible.call:
    - system.ping:
        - data: "Hello from Salt"
```

```
# salt-call --local ansible.system.ping data=”Hello from Salt”
'ansible.system.ping' is not available.
```

```
# salt salt-master* ansible.help system.ping
salt-master:
    ----------
    Available sections on module "system.ping":
        - module
        - version_added
        - short_description
        - options
        - seealso
        - author
    Description:
        - A trivial test module, this module always returns C(pong) on successful contact. It does not make sense in playbooks, but it is useful from C(/usr/bin/ansible) to verify the ability to login and that a usable Python is configured.
        - This is NOT ICMP ping, this is just a trivial test module that requires Python on the remote-node.
        - For Windows targets, use the M(win_ping) module instead.
        - For Network targets, use the M(net_ping) module instead.

```

**Comment by sagetherage** [comment]
@slinderud we are still in planning for the next release cycle and hope to finish that up this week and then move many issues like this into releases/milestones - thank you for the reminder

**Comment by slinderud** [comment]
> @slinderud we are still in planning for the next release cycle and hope to finish that up this week and then move many issues like this into releases/milestones - thank you for the reminder

Thank you :) Really looking forward to get this working!

**Comment by whytewolf** [comment]
so looking into this. this does work. the problem is it isn't listed as system.ping. just like you don't call it system.ping in ansible. it is just ansible.ping. 

```
sudo salt-call --local sys.list_functions ansible | grep ping
    - ansible.cisco.ios.ios_ping
    - ansible.cisco.nxos.nxos_igmp_snooping
    - ansible.cisco.nxos.nxos_ping
    - ansible.community.general.pingdom
    - ansible.community.general.postgresql_ping
    - ansible.community.network.icx_ping
    - ansible.community.network.pn_igmp_snooping
    - ansible.community.postgresql.postgresql_ping
    - ansible.fortinet.fortios.fortios_firewall_shaping_policy
    - ansible.fortinet.fortios.fortios_firewall_shaping_profile
    - ansible.fortinet.fortios.fortios_switch_controller_igmp_snooping
    - ansible.junipernetworks.junos.junos_ping
    - ansible.netapp_eseries.santricity.na_santricity_lun_mapping
    - ansible.netapp_eseries.santricity.netapp_e_lun_mapping
    - ansible.netcommon.net_ping
    - ansible.openstack.cloud.federation_mapping
    - ansible.openstack.cloud.federation_mapping_info
    - ansible.openstack.cloud.os_keystone_mapping
    - ansible.openstack.cloud.os_keystone_mapping_info
    - ansible.ping
    - ansible.vyos.vyos.vyos_ping
    - ansible.windows.win_ping
❯ sudo salt-call --local ansible.ping
local:
    ----------
    ping:
        pong

```

hope this helps.

**Comment by peter-fe** [comment]
@whytewolf , wow thx for coming back in this, it's quite while when I opened this.
If iI memorize correctly calling ping was just an example for the broken output forwarding from Ansible called by Salt.
It worked flawlessly and then broke withe some release in 2020, so I tried to create a minimal test case to reproduce (and maybe not a good one, ...).
I have not used Salt for some years now, so maybe also the initial issue of the missing output from Ansible through Salt is long fixed.

I'd like to thank you for picking it up though, but would close it as I can't verify if the issue is still there, or what the  original error case was
